libraries/node-core-library/src/Import.ts

```diff
@@ -83,4 +128,105 @@ export class Import {
   }
+
+  /**
+   * Resolves a path in a package, relative to another path.
+   */
+  public static resolve(options: IImportResolveOptions): string {
+    const { resolvePath } = options;
+
+    if (nodeJsPath.isAbsolute(resolvePath)) {
+      return resolvePath;
+    }
+
+    let normalizedRootPath: string = FileSystem.getRealPath(options.baseFolderPath);
+
+    if (resolvePath.startsWith('.')) {
+      // This looks like a conventional relative path
+      return nodeJsPath.resolve(normalizedRootPath, resolvePath);
+    }
+
+    normalizedRootPath =
+      PackageJsonLookup.instance.tryGetPackageFolderFor(normalizedRootPath) || normalizedRootPath;
+
+    let slashAfterPackageNameIndex: number;
+    if (resolvePath.startsWith('@')) {
+      // This looks like a scoped package name
+      slashAfterPackageNameIndex = resolvePath.indexOf('/', resolvePath.indexOf('/') + 1);
+    } else {
+      slashAfterPackageNameIndex = resolvePath.indexOf('/');
+    }
+
+    let packageName: string;
+    let pathInsidePackage: string | undefined;
+    if (slashAfterPackageNameIndex === -1) {
+      // This looks like a package name without a path
+      packageName = resolvePath;
+    } else {
+      packageName = resolvePath.substr(0, slashAfterPackageNameIndex);
+      pathInsidePackage = resolvePath.substr(slashAfterPackageNameIndex + 1);
+    }
+
+    if (options.includeSystemModules === true) {
+      if (!Import._builtInModules) {
+        Import._builtInModules = new Set<string>(nodeModule.builtinModules);
+      }
+
+      // First, check resolvePath because some built-in modules have more than one slash in their name
+      if (Import._builtInModules.has(resolvePath)) {
+        return resolvePath;
+      } else if (Import._builtInModules.has(packageName)) {
+        if (pathInsidePackage !== undefined) {
+          throw new Error(
+            `The package name "${packageName}" resolved to a NodeJS system module, but the ` +
+              `path to resolve ("${resolvePath}") contains a path inside the system module, which is not allowed.`
+          );
+        }
+
+        return packageName;
+      }
+    }
+
+    let resolvedPackagePath: string | undefined;
+    if (options.allowSelfReference === true) {
+      // See if we're trying to resolve to the current package
+      const ownPackageJsonPath: string | undefined = PackageJsonLookup.instance.tryGetPackageJsonFilePathFor(
+        normalizedRootPath
+      );
+      if (
+        ownPackageJsonPath &&
+        PackageJsonLookup.instance.loadPackageJson(ownPackageJsonPath).name === packageName
+      ) {
+        resolvedPackagePath = nodeJsPath.dirname(ownPackageJsonPath);
+      }
+    }
+
+    if (!resolvedPackagePath) {
+      try {
+        resolvedPackagePath = nodeJsPath.dirname(
+          Resolve.sync(
+            // Append a slash to the package name to ensure `resolve.sync` doesn't attempt to return a system package
+            `${packageName}/`,
```

Comment:
I don't see anything ever getting added to the `_builtInModules` set. (?)

According to the `resolve` package maintainer, appending a slash or not is the magic way to control whether it considers system modules or not.  He chose this design because it is how Node.js `require()` behaves.

However, I find that highly counterintuitive.  In 99% of cases, `Import.resolve()` will be resolving NPM packages, and it would be wrong to match a system module. 

@iclanton Thus I'm thinking the design can be like this:

- by default if `includeSystemModules=false`, we append a trailing slash ONLY if `resolvePath` does not already contain a slash
- if `includeSystemModules=true` then we do not append a slash.  Thus `resolvePath="fs"` would return the system path. I'm undecided about what to do with ``resolvePath="fs/"` in this case.  Faithfully emulate Node.js? Or assume it is a mistake an report an error?  Not sure what's better.

Reply:
`_builtInModules` is set on [line 172](https://github.com/microsoft/rushstack/pull/2095/files/8fe1fb927b899284053cf3c555acd5d6cad7fcd9#diff-cf89596b0e435a3be751cbc2b37072baR171-R173)

`resolve` relies on a janky [hardcoded list](https://github.com/browserify/resolve/blob/master/lib/core.json) of "core" modules, while this solution gets that list from the runtime. I prefer my solution.